docs/how-to/deploy/deploy-to-production.mdx

```diff
@@ -132,3 +132,47 @@ To achieve high throughput and low latency:
 - **Optimize Base Policy**: Tailor the base policy to your specific use case. For example, disabling ReBAC checks (if not used) can speed up policy execution (this requires our tech support).
   With these optimizations, you can expect several thousand RPS with less than 10ms latency per request.
+
+
+## PDP Offline Mode
+
+The PDP relies on Permit's cloud for fetching configuration, loading policies, and keeping those up to date.
+Despite the high availability of Permit's cloud, there still might be cases when the PDP can't reach it.
+For extra resilience, the PDP can be configured to work in offline mode, where it will use a local backup of the configuration and policies.
+
+### How it works
+When the PDP is started in offline mode, it will try to fetch the configuration from Permit's cloud as usual.
+After `PDP_CONFIG_FETCH_MAX_RETRIES` failed attempts, the PDP will switch to offline mode and use the backup configuration file if present.
+Once PDP is up and running - it won't try to fetch the configuration again from cloud, unless it's restarted.
+
+For backing up the policies (both data & rules loaded to policy store), PDP internally uses [OPAL's offline mode](https://docs.opal.ac/getting-started/running-opal/run-opal-client/opa-runner-parameters/#policy-store-backup) 
+(no need to explicitly enable OPAL's offline mode, PDP would do it automatically).
+OPAL uses policy data from the backup file to initialy load the policy store, but it never gives up on syncing with cloud - so once Permit cloud is reachable again, the most updated policy data could be fetched.
+Policy data backup is done periodically (to the same directory as the configuration backup file), and also on a graceful shutdown of the PDP.
+
+
+### Configuring Offline Mode
+
+To enable the PDP offline mode, use the `PDP_ENABLE_OFFLINE_MODE` environment variable.
+You would also have to mount a volume or a host directory to the PDP container at `/app/backup`, where the backup files will be persisted.
+When PDP offline mode is enabled, the PDP will automatically enable OPAL's offline mode, and will use the backup files from the mounted volume.
+
+```sh
+docker run -it -p 7766:7000 -p 8181:8181 
+  --env PDP_API_KEY=<YOUR_API_KEY> \
+  --env PDP_ENABLE_OFFLINE_MODE=true \
+  -v <HOST_BACKUP_DIR_OR_NAMED_VOLUME>:/app/backup
+  permitio/pdp-v2:latest
+```
+
+There are serverl other advanced configuration options you can use:
+* `PDP_OFFLINE_MODE_BACKUP_DIR` Sets the directory path of the PDP's configuration backup file (default "/app/backup")
+* `PDP_OFFLINE_MODE_BACKUP_FILENAME` Sets the file name of the PDP's configuration backup file (default "pdp_cloud_config_backup.json")
+* `PDP_OFFLINE_MODE_POLICY_BACKUP_FILENAME` Sets the file name of the PDP's policy store backup file within the backup dir. This replaces OPAL's `OPAL_STORE_BACKUP_PATH`. (default "policy_store_backup.json")
```

Comment:
What about data backup file ?

Reply:
Cool - so maybe let's just clarify that

Reply:
I'll try to make it clearer.
Although I'm always frustrated and confused about needing a word to describe all data loaded to OPA (which contains both data & policy). sometimes I call it "Policy store data" but that apparently isn't clear either.

Reply:
How is it now?